src/plugins/data_source/common/data_sources/types.ts

```diff
@@ -6,6 +6,7 @@
 import { SavedObjectAttributes } from 'src/core/types';
 
 export interface DataSourceAttributes extends SavedObjectAttributes {
+  id?: string;
```

Comment:
I don't think id should be added here, as it is part of type `SavedObjectAttributes`

Reply:
file: test_connection.ts & line: 41 .. is the reason I added this as optional field.